Bottom: 90262b25e58a9c80e1b89ff7ef39e7531f40b961
Top:    12be089eab1fc3efb16038d1df1d560b204eaf44
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2013-12-31 23:31:27 -0500

Refresh of zarlink

---

diff --git a/package/kernel/linux/modules/sound.mk b/package/kernel/linux/modules/sound.mk
index 777d274..2bc3ed1 100644
--- a/package/kernel/linux/modules/sound.mk
+++ b/package/kernel/linux/modules/sound.mk
@@ -280,3 +280,41 @@ define KernelPackage/pcspkr/description
 endef
 
 $(eval $(call KernelPackage,pcspkr))
+
+define KernelPackage/sound-soc-rt5350
+  TITLE:=RT5350 SoC support
+  KCONFIG:=\
+	CONFIG_SND_SOC_RT5350_PCM
+  FILES:= \
+	$(LINUX_DIR)/sound/soc/fsl/snd-soc-rt5350-pcm.ko
+  AUTOLOAD:=$(call AutoLoad,56,snd-soc-rt5350-pcm)
+  DEPENDS:=@TARGET_rt5350 +kmod-sound-soc-core
+  $(call AddDepends/sound)
+endef
+
+define KernelPackage/sound-soc-rt5350/description
+ Support for RT5350 Platform sound (pcm)
+endef
+
+$(eval $(call KernelPackage,sound-soc-rt5350))
+
+
+define KernelPackage/sound-soc-rt5350-le89156
+  TITLE:=RT5350 SoC support for Zarlink LE89156
+  KCONFIG:=SND_RT5350_SOC_LE89156
+  FILES:=\
+	$(LINUX_DIR)/sound/soc/codecs/snd-soc-le89156.ko \
+	$(LINUX_DIR)/sound/soc/codecs/snd-soc-rt5350-le89156.ko
+  AUTOLOAD:=$(call AutoLoad,57,snd-soc-rt5350-le89156)
+  DEPENDS:=@TARGET_rt5350 +kmod-snd-soc-rt5350
+  $(call AddDepends/sound)
+endef
+
+define KernelPackage/sound-soc-rt5350-le89156/description
+ Support for RT5350 Platform sound Zarlink LE89156 codec
+endef
+
+$(eval $(call KernelPackage,sound-soc-rt5350-le89156))
+
+
+
diff --git a/target/linux/ramips/patches-3.10/0320-rt5350-zarlink.patch b/target/linux/ramips/patches-3.10/0320-rt5350-zarlink.patch
index 3cc7882..ef6b33e 100644
--- a/target/linux/ramips/patches-3.10/0320-rt5350-zarlink.patch
+++ b/target/linux/ramips/patches-3.10/0320-rt5350-zarlink.patch
@@ -1856,3 +1856,13 @@
 +MODULE_DESCRIPTION("Ralink RT5350 SoC PCM driver");
 +MODULE_LICENSE("GPL");
 +MODULE_ALIAS("platform:rt5350-pcm");
+--- a/sound/soc/codecs/Makefile
++++ b/sound/soc/codecs/Makefile
+@@ -32,6 +32,7 @@ snd-soc-dmic-objs := dmic.o
+ snd-soc-isabelle-objs := isabelle.o
+ snd-soc-jz4740-codec-objs := jz4740.o
+ snd-soc-l3-objs := l3.o
++snd-soc-le89156-objs := le89156.o
+ snd-soc-lm4857-objs := lm4857.o
+ snd-soc-lm49453-objs := lm49453.o
+ snd-soc-max9768-objs := max9768.o
